### PR TITLE
fixed an issue with an export and added support for qb-jobs

### PR DESCRIPTION
**Pull request by r0adra93**
Ready for qb-jobs

**Describe Pull request**
These qb-jobs support items need to be tested against the current build of QB-Core to ensure that it works as intended then once it is approved then it should be put into the core as it works.

Once qb-jobs is ready all these items need to be tested against qb-jobs to ensure they are working as expected with qb-jobs.

Then once qb-jobs is released as long as the server operators have been keeping their servers up-to-date they can just drop qb-jobs in and it will work.

Added support for qb-jobs

removed a warning item in VS.Code that stated table.clone was invalid.

If your PR is to fix an issue mention that issue here

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? yes (Be honest)
- Does your code fit the style guidelines? [yes 
- Does your PR fit the contribution guidelines? yes

### CODE REVIEW

**Review by KamuiKody — CHANGES_REQUESTED** [review]
when will it ever touch "not availableJobs" at line 20? Im pretty sure that is unnecessary but the rest of it looks ok i will have to check out your export for CityHallJobPop()

**Comment by r0adra93** [comment]
> when will it ever touch "not availableJobs" at line 20? Im pretty sure that is unnecessary but the rest of it looks ok i will have to check out your export for CityHallJobPop()

It happened a few times testing with qb-jobs.

**Comment by KamuiKody** [comment]
It needs this table to populate city hall though i would then think that is a bug if the table shows up nil

**Comment by r0adra93** [comment]
I see it now, it was a remnant of getting AddCityJob to work when the availableJobs table was empty.

AddCityJob does not work as expected though. 

The variable does not update outside of the function once the code is loaded. 

I know this because everything I did to export into AddCityJob the menu kept coming up blank.

So, I had to create an export for qb-jobs and do what I did above to get the menu to populate. 

Now, when people create new jobs and use qb-jobs to create said jobs qb-cityhall will be automatically updated as long as the inCityHall flag is populated.

That function is a test when qb-jobs is released.

File: client/main.lua [hunk 1]
```diff
@@ -199,8 +199,19 @@ local function deletePeds()
...
+    QBCore.Functions.TriggerCallback('qb-cityhall:server:receiveJobs', function(result)
+        SendNUIMessage({
+            action = 'setJobs',
+            jobs = result
+        })
+    end)
+end
 
+-- Events
+RegisterNetEvent('cityhall:Client:AddCityJob', function(jobName,label)
```
Comment (tom-osborne) [inline client/main.lua:212]: Incorrect event name, should be `qb-cityhall`

**Review by tom-osborne — CHANGES_REQUESTED** [review]

File: client/main.lua (hunk 1 above)
Comment (r0adra93) [reply client/main.lua]: Done

File: server/main.lua [hunk 2]
```diff
@@ -16,6 +19,7 @@ local function AddCityJob(jobName, label)
         return false, "already added"
     else
         availableJobs[jobName] = label
+        TriggerClientEvent('qb-cityhall:Client:AddCityJob',-1)
```
Comment (charj) [inline server/main.lua:22]: Missing whitespace before `-1`

Comment (r0adra93) [reply server/main.lua:22]: Whitespace is not needed.

Comment (charj) [reply server/main.lua:22]: It's not a requirement, but it's good to adhere to a consistent code style

Comment (charj) [reply server/main.lua:22]: In your above code

`RegisterNetEvent('qb-cityhall:Client:AddCityJob', function()
    getJobs()
end)`

vs 

`RegisterNetEvent('qb-cityhall:Client:AddCityJob',function()
    getJobs()
end)`

**Comment by charj** [comment]
Code is looking good @r0adra93 - I need to test it in order to approve, will do that later.

**Comment by tom-osborne** [comment]
> Code is looking good @r0adra93 - I need to test it in order to approve, will do that later.

@charj test this PR too please - I think this worked when I did it originally https://github.com/qbcore-framework/qb-cityhall/pull/81

File: html/app.js [hunk 3]
```diff
@@ -106,6 +107,7 @@ $(".request-identity-button").click(function(e){
 $(document).on("click", ".job-page-block", function(e){
     e.preventDefault();
     selectedJobId = $(this).data('job');
+//    selectedJobId["application"] = $(this).data('application')
```
Comment (charj) [inline html/app.js:110]: Do we need this?

Comment (r0adra93) [reply html/app.js:110]: the variable application does not exist in the file.

**Review by zwkf — APPROVED** [review]

**Pull request merged**